### PR TITLE
[BPF] fix forwarding and NATing of ICMP related messages

### PR DESCRIPTION
**Pull request by tomastigera**
When a service (nodeport) is accessed from outside the cluster and traffic from backend generates ICMP error outside the cluster, we need to deliver the ICMP error (e.g. MTU too big) back to the backend workload so that it can adjust. These packets were not sent back into the vxlan tunnel neither we did DNAT on them if they were.

fixes https://github.com/projectcalico/calico/issues/8854

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: fixed forwarding, NATing and checksuming of related ICMP traffic (icmp errors)
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by fasaxc — CHANGES_REQUESTED** [review]
Couple of tiny nits and a question on the mark pattern.  Thanks for all the tests, exactly the sort of function that it's easy to regress without good tests.

File: felix/fv/bpf_test.go [hunk 1]
```diff
@@ -3965,6 +3969,106 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
...
+							})
+
+							It("should have connectivity to service backend", func() {
+								tcpdump := tc.Felixes[0].AttachTCPDump(w[0][0].InterfaceName)
+								tcpdump.SetLogEnabled(true)
+								tcpdump.AddMatcher("mtu-1300", regexp.MustCompile("mtu 1300"))
+								tcpdump.Start("-vvv", "icmp", "or", "icmp6")
+								defer tcpdump.Stop()
+
+								ipRouteFlushlushCache := []string{"ip", "route", "flush", "cache"}
```
Comment (sridhartigera) [inline felix/fv/bpf_test.go:4043]: ```ipRouteFlushCache```?

File: felix/bpf-gpl/bpf.h [hunk 2]
```diff
@@ -190,14 +190,18 @@ enum calico_skb_mark {
...
-        */
-       CALI_SKB_MARK_TO_NAT_IFACE_OUT        = 0x41000000,
-       /* CALI_SKB_MARK_FROM_NAT_IFACE_OUT signals to the next hop that the packet passed
-	* through bpfnatout so that it can set its conntrack correctly.
-	*/
-       CALI_SKB_MARK_FROM_NAT_IFACE_OUT      = 0x81000000,
+	/* CALI_SKB_MARK_RELATED_RESOLVED mean it is related traffic, we already
+	 * know that and we have resolved NAT etc.
+	 */
+	CALI_SKB_MARK_RELATED_RESOLVED        = 0x11000000,
```
[the remark this answers is not in the document]
Comment (tomastigera) [reply felix/bpf-gpl/bpf.h:196]: Noooooooooooooo :man_facepalming: I was sooo happy that I had one bit

[the remark this answers is not in the document]
Comment (tomastigera) [reply felix/bpf-gpl/bpf.h]: Ohhh there seems to be one more left! :tada:

**Review by sridhartigera — APPROVED** [review]
LGTM

**Comment by fasaxc** [comment]
CI failing with a timeout in the BPF tests?

**Comment by tomastigera** [comment]
> CI failing with a timeout in the BPF tests?

That is unfortunate, but unrelated as it predates this patch :-/

**Comment by tomastigera** [comment]
some unrelated flakes

**Pull request merged**